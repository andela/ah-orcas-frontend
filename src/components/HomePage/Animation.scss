.card {
  -ms-transform: perspective(1000px);
  -moz-transform: perspective(1000px);
  -ms-transform-style: preserve-3d;
  -moz-transform-style: preserve-3d;
  perspective: 1000;
  transition: all 0.5s cubic-bezier(0.05, 0.05, 0.05, 0.05);
}
