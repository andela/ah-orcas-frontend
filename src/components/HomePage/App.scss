//color variables
$border-color: #e9eaea;
$btn-color: #3b5998;
$focus-color: #007bff;
$font-color: #424242;
$primary-color: #f5f5f5;
$solid-primary: #000;
$webkit: #3e3e40;

$lightprimarycolor: #fff;
$darkcolor: #888;
$primarycolor: #fff;
$color-primary-dark: #fff;
$color-accent: #fff;
$black: #000;
$navbar-height: 100%;

.myform {
  border-radius: 2em;
  border-style: solid;
  display: block;
  margin-left: 23em;
  margin-right: 7em;
  margin-top: 6em;
  padding: 40px;
  width: 80%;
}

* {
  margin: 0;
  padding: 0;
}

.body {
  background: primary-color;
  font-family: roboto;
  user-select: none;
}


.signup {
  cursor: pointer;
  font-size: 22px;
  font-weight: bolder;
  text-align: center;
  text-transform: uppercase;
  width: 88%;
}

.signup-form {
  box-decoration-break: clone;
  box-shadow: 2em;
  box-sizing: border-box;
}

.input {
  border: 2px;
  box-sizing: border-box;
  color: webkit;
  font-size: 14px;
  margin-bottom: 25px;
  outline: none;
  padding: 20px;
  width: 100%;
}


span {
  a {
    color: solid-primary;
    text-decoration: none;
}
}

.mybtn {
  background-color: btn-color;
  border-radius: 0.25em;
  display: inline-block;
  font-weight: 400;
  padding: 0.375rem 0.75rem;
  text-align: center;
  width: 100%;
}

.myalert {
  background-color: primary-color;
  border: solid transparent;
  border-radius: 2em;
  color: focus-color;
  margin-bottom: 1rem;
  padding: 0.75rem 1.25rem;
  position: relative;
}

.articlecard {
  font-family: 'Bitter', serif;
  margin-left: 1rem;
  margin-right: 1rem;
  margin-top: 2rem;
}

.intro-text {
  color: $black;
  font-size: 5em;
  opacity: 1;
  padding-left: 10%;
  padding-top: 5%;
}

.title {
  color: $black;
  text-align: center;
}

.card:hover {
  box-shadow: 3px 3px 3px;
  z-index: 1;
}

.description {
  font-size: 60rem;
}

.top {
  border-radius: 10px;
  box-shadow: 5px 5px 5px $darkcolor;
  height: 200px;
  margin: 0;
  margin-bottom: 2%;
  opacity: 0.8;
  position: relative;
  width: 100%;
  z-index: 9;
}

.btn:hover {
  box-shadow: 1px 1px 1px $darkcolor;
}

.card {
  border-radius: 10px;
}

.article-content {
  border-radius: 10px;
  box-shadow: 1px 1px 1px $darkcolor;
  margin-bottom: -1%;
  padding: 1%;
}

.article-container {
  background-color: $primarycolor;
  padding-bottom: 5%;
  padding-top: 5%;
}

.container-text {
  background-color: $lightprimarycolor;
  border-radius: 10px;
  margin-bottom: 1%;
  margin-top: 1%;
  padding: 1%;
  text-align: left;
}

.text {
  margin-left: 1%;
}

.time {
  float: right;
  margin-left: 1%;
}

.clear {
  clear: both;
}

.username {
  float: left;
  margin-left: 1%;
}

.userimage {
  float: left;
  margin: 1%;
  width: 5%;
}

h1, h3, h4, h5, h6, p, a, ul, span, li, img { 
  font-weight: 300;
  margin: 0; 
  padding: 0;   
}

h1,h3,h4,h5,h6 { 
  line-height: 1.5; 
}

$color-paragraph: #444;

p { 
  color: $color-paragraph; 
  font-family: 'Maven Pro', sans-serif; 
  font-weight: 400; 
  line-height: 1.6; 
}

h1 { 
  font-size: 4em; 
  line-height: 1; 
}

h3 { 
  font-size: 2em; 
}

h4 { 
  font-size: 1.2em; 
}

h5 { 
  font-size: 1.1em; 
}

h6 { 
  font-size: .9em; 
  letter-spacing: 1px; 
}

a, button { 
  color: inherit; 
  display: inline-block; 
  text-decoration: none; 
  transition: all .3s; 
}

$color-focus: #498BF9;
a:focus, a:active, a:hover,
button:focus, button:active, button:hover { 
  color: $color-focus; 
  text-decoration: none; 
}

b { 
  font-weight: 500; 
}

img { 
  width: 100%; 
}

li { 
  list-style: none; 
  display: inline-block; 
}

span { 
  display: inline-block; 
}

.section { 
  padding: 70px 0 40px; 
}

.section .heading{ 
  padding-bottom: 70px; 
}

.card { 
  background: transparent; 
  border: 0; 
}

.blog-area .row div[class^="col"] { 
  margin-bottom: 30px; 
}

$color-blog-area-background: #EDF3F3;

.blog-area { 
  background: $color-blog-area-background; 
  text-align: center; 
}

$color-blog-area-box-shadow: rgba(0,0,0,.1);
$color-blog-area-background: #fff;
$color-blog-area-border: #ddd;

.blog-area .single-post { 
  background: $color-blog-area-background;
  border: 1px solid $color-blog-area-border; 
  box-shadow: 0 0 5px $color-blog-area-box-shadow; 
  height: 100%; 
  overflow: hidden; 
  padding-bottom: 45px;
  position: relative; 
}

.blog-area .single-post .title { 
  padding: 20px 30px 30px; 
}

$color-single-post-box-shadow: rgba(0,0,0,.2);

.single-post .avatar { 
  border: 7px solid $color-blog-area-background; 
  border-radius: 100px; 
  box-shadow: 0 0 10px $color-single-post-box-shadow; 
  height: 70px; 
  margin: 0 auto; 
  margin-top: -40px; 
  overflow: hidden; 
  position: relative; 
  width: 70px; 
  z-index: 10;
}

.single-post .post-footer { 
  bottom: 0;
  left: 0; 
  position: absolute; 
  right: 0; 
  text-align: center; 
}

.single-post .post-footer > li { 
  background: $color-blog-area-background; 
  border-right: 1px solid $lightprimarycolor; 
  display: inline-block; 
  width: 33.33%; 
}

.single-post .post-footer > li:first-child { 
  float: left; 
}

.single-post .post-footer > li:last-child { 
  border: 0; 
  float: right; 
}

.single-post .post-footer > li > a { 
  line-height: 45px; 
  width: 100%; 
}

.single-post .post-footer i { 
  display: inline-block; 
  font-size: 1.1em; 
  margin-right: 10px; 
  opacity: .6; 
}

.post-style-1 .blog-image { 
  max-height: 200px; 
  overflow: hidden; 
}
