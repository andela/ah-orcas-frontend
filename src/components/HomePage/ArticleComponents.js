import React from 'react';
import 'bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Link } from 'react-router-dom';
import baseUrl from '../../constants/url';
import { ReadMore, WELCOME_NOTE } from '../../constants/articles';

const URL = `${process.env.URL}/article/`;


export const Title = props => (
  <div>
    <a href={URL} className="title">
      {
      props.title
    }
    </a>
  </div>
);

export const ArticleContainer = props => (
  <div className="card-container articlecard">
    <div className="card-columns">
      {props.newArticles}
    </div>
  </div>
);

export const Image = props => (
  <img className="card-img-top" alt="" src="https://picsum.photos/800/200?random" />
);


export const Description = (props) => {
  const { slug } = props;
  return (
    <div className="lead">
      <p className="text-left description">
        {props.description}
        <Link to={`/article/detail/${slug}`}>
          <button onClick={() => (localStorage.setItem('slug', slug))} type="button" className="btn" style={{ background: '#009688' }}>
            {ReadMore}
          </button>
        </Link>
      </p>
    </div>
  );
};

export const Userimage = props => (
  <img alt="" className="img-rounded userimage" src={props.userimage} />
);
export const Username = props => (
  <a className="small username title" href={URL}>
    {props.username}
  </a>
);

export const TimeCreated = props => (
  <span className="small time">{props.time}</span>
);

class Intro extends React.Component {
  stat={
    display: 'block',
  }

  componentDidMount() {
    setTimeout(
      () => {
        this.setState({ display: 'none' });
      },
      8000,
    );
  }

  render() {
    return (
      <div className="top" style={this.state}>
        <h1 className="intro-text anim-typewriter">{WELCOME_NOTE}</h1>
      </div>
    );
  }
}


export const IntroComponent = Intro;

export default Title;

// export default connect(mapStateToProps, mapDispatchToProps)(View);
