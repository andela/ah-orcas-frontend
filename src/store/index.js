import { createStore, applyMiddleware } from 'redux';
import { createLogger } from 'redux-logger';
import thunk from 'redux-thunk';

import allReducers from '../reducers';

const logger = createLogger();
export const store = createStore(allReducers, applyMiddleware(logger, thunk));
export default store;
