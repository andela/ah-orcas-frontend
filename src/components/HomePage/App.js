import React, { Component } from 'react';
import './App.scss';
import './Animation.scss';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import propTypes from 'prop-types';
import { getAllArticles } from '../../actions/articles.action';
import ArticleCard, { MyLoader } from './containers';
import { USER_IMAGE } from '../../constants/articles';


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      page: 1,
    };

    this.handleLoadMore = this.handleLoadMore.bind(this);
    this.handleLoadPrev = this.handleLoadPrev.bind(this);
  }

  componentDidMount() {
    const { fetchAllArticles } = this.props;
    const { page } = this.state;
    fetchAllArticles(page);
  }

  handleLoadMore() {
    const { page } = this.state;
    const { fetchAllArticles, articles } = this.props;
    // if the page === page_size, there could be more articles
    if (articles.length === 12) {
      this.setState({ page: page + 1 });
    }
    fetchAllArticles(page + 1);
  }

  handleLoadPrev() {
    const { page } = this.state;
    const { fetchAllArticles } = this.props;
    if (page === 1) {
      this.setState({ page: 1 });
    } else {
      this.setState({ page: page - 1 });
    }
    fetchAllArticles(page - 1);
  }

  render() {
    const { page } = this.state;
    const { articles } = this.props;
    const newArticles = articles.map(article => (
      <ArticleCard
        key={article.id}
        slug={article.slug}
        title={article.title}
        article={article}
        username={article.author.username}
        timecreated={article.timecreated}
        description={article.description}
        userimage={USER_IMAGE}
        comments={article.comments}
        likes={article.likes}
        dislikes={article.dislikes}
      />
    ));
    return (
      <section className="blog-area section">
        <div className="container">
          {
            articles.length !== 0
              ? (
                <div>
                  <div className="row">
                    {newArticles}
                  </div>
                  <ul className="pagination pagination-lg">
                    {
                      page !== 1
                        ? (
                          <li className="page-item"><a className="page-link" href onClick={this.handleLoadPrev}>Prev</a></li>
                        )
                        : (
                          ''
                        )
                    }
                    {
                      articles.length >= 12
                        ? (
                          <li className="page-item"><a className="page-link" href onClick={this.handleLoadMore}>Next</a></li>
                        )
                        : (
                          ''
                        )
                    }
                  </ul>
                </div>
              )
              : (
                <div className="row">
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                  <MyLoader className="col-lg-4 col-md-6 d-flex align-items-stretch" />
                </div>
              )
          }
        </div>
      </section>
    );
  }
}

App.propTypes = {
  fetchAllArticles: propTypes.func,
  articles: propTypes.array,
};

App.defaultProps = {
  fetchAllArticles: propTypes.func,
  articles: propTypes.array,
};

function mapStateToProps(state) {
  return {
    articles: state.articles,
  };
}


function mapDispatchToProps(dispatch) {
  return {
    fetchAllArticles: bindActionCreators(getAllArticles, dispatch),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
