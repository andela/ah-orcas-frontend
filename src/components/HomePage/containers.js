import React from 'react';
import ContentLoader from 'react-content-loader';
import './App.scss';
import './Animation.scss';
import 'bootstrap';
import propTypes from 'prop-types';
import 'bootstrap/dist/css/bootstrap.min.css';

const profileImage = require('../../assets/images/profile.png');
const articleImage = require('../../assets/images/cup.jpg');

const ArticleCard = (props) => {
  const { slug, username, title, likes, dislikes, comments } = props;
  const profileLink = `/profile/${username}`;
  const articleLink = `/article/detail/${slug}`;

  return (
    <div className="col-lg-4 col-md-6 d-flex align-items-stretch">
      <div className="card">
        <div className="single-post post-style-1">
          <div className="blog-image">
            <img src={articleImage} alt="article pic" />
          </div>
          <a className="avatar username" href={profileLink}>
            <img src={profileImage} alt="user profile pic" />
          </a>
          <div className="clear" />
          <div className="blog-info">
            <h4 className="title">
              <a href={articleLink}><b>{title}</b></a>
              <br />
              <span className="username"><a href={profileLink}><b>{username}</b></a></span>

            </h4>
            <ul className="post-footer">
              <li>
                <a href={articleLink}>
                  <i className="glyphicon glyphicon-thumbs-up" />
                  {likes}
                </a>

              </li>
              <li>
                <a href={articleLink}>
                  <i className="glyphicon glyphicon-thumbs-down" />
                  {dislikes}
                </a>

              </li>
              <li>
                <a href={articleLink}>
                  <i className="glyphicon glyphicon-comment" />
                  {comments}
                </a>

              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  );
};

ArticleCard.propTypes = {
  userimage: propTypes.string,
  timecreated: propTypes.string,
  description: propTypes.string,
  username: propTypes.string,
  title: propTypes.string,
  image: propTypes.string,
  slug: propTypes.string,
  likes: propTypes.number,
  dislikes: propTypes.number,
  comments: propTypes.number,
};

ArticleCard.defaultProps = {
  userimage: '',
  timecreated: '',
  description: '',
  username: '',
  title: '',
  image: '',
  slug: '',
  likes: 0,
  dislikes: 0,
  comments: 0,
};


export default ArticleCard;

export const MyLoader = props => (
  <ContentLoader
    height={500}
    width={400}
    speed={2}
    primaryColor="#f3f3f3"
    secondaryColor="#ecebeb"
    {...props}
  >
    <circle cx="200" cy="252" r="30" />
    <rect x="152.8" y="299" rx="4" ry="4" width="100" height="13" />
    <rect x="179" y="332.83" rx="4" ry="4" width="50" height="14.4" />
    <rect x="1" y="70" rx="5" ry="5" width="400" height="176" />
    <rect x="6" y="385" rx="4" ry="4" width="50" height="14.4" />
    <rect x="179" y="385" rx="4" ry="4" width="50" height="14.4" />
    <rect x="340" y="385" rx="4" ry="4" width="50" height="14.4" />
  </ContentLoader>
);
