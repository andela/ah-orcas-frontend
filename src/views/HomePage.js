import React from 'react';
import AuthLogin from '../components/LoginPage/App';
import ResetPasswordComp from '../components/ResetPassword/ResetPassword';
import RegisterPage from '../components/SignUpPage/registerComponent';
import Profile from '../components/ProfilePage/Profile';
import HomePage from '../components/HomePage/App';
import ArticleInput from '../components/ArticleCreate/App';
import Article from '../components/ArticleView/App';
import Update from '../components/ArticleUpdate/App';
import { NavBarComponent, FooterComponent } from '../components/NavBar/App';
import RateArticle from '../components/RateArticle/rateArticleComponent';
import Reports from '../components/reports/reports';


export const HomePageView = () => (
  <HomePage className="grid" />
);

export const LoginPage = ({ ...props }) => (
  <AuthLogin {...props} />
);

export const Header = () => (
  <NavBarComponent />
);

export const Footer = () => (
  <FooterComponent />
);

export const Register = () => (
  <RegisterPage />
);

export const ResetPassword = () => (
  <ResetPasswordComp />
);

export const ArticleInputComponents = () => (
  <ArticleInput />
);

export const ArticleComponent = () => (
  <Article />
);

export const UpdateArticleComponent = () => (
  <Update />
);

export const ProfilePage = ({ ...props }) => (
  <Profile {...props} />
);

export const Rate = () => (
  <RateArticle />
);

export const reportArticle = () => (
  <Reports />
);
