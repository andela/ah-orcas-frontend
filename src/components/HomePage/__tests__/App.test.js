import React from 'react';
import { shallow } from 'enzyme';
import store from '../../../store';
import HomePage from '../App';

function setup() {
  const props = {
    articles: [],
  };
  const enzymeWrapper = shallow(<HomePage {...props} store={store} />);
  return {
    props,
    enzymeWrapper,
  };
}

describe('homepage component', () => {
  describe('homepage', () => {
    it('should render self and subcomponents', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
      expect(enzymeWrapper.find('div').length).toBe(0);
    });
  });
});
