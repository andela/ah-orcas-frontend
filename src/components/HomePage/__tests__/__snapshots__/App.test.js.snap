// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login components login shou ld render self and subcomponents 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(App)
    login={[MockFunction]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "articles": Array [],
      "fetchAllArticles": [Function],
      "login": [MockFunction],
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "articles": Array [],
        "fetchAllArticles": [Function],
        "login": [MockFunction],
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
